crud/common/base_plan.lua

```diff
@@ -0,0 +1,65 @@
+local errors = require('errors')
+
+local compare_conditions = require('crud.compare.conditions')
+local utils = require('crud.common.utils')
+local dev_checks = require('crud.common.dev_checks')
+
+local IndexTypeError = errors.new_class('IndexTypeError', {capture_stack = false})
+
+local base_plan = {}
+
+function base_plan.new(space, conditions)
+    dev_checks('table', '?table')
+
+    local scan_index
+    local scan_iter
+    local scan_value
+    local scan_condition_num
+
+    conditions = conditions ~= nil and conditions or {}
+
+    local space_indexes = space.index
+    local space_format = space:format()
+
+    if conditions == nil then -- also cdata<NULL>
+        conditions = {}
+    end
+
+    -- search index to iterate over
+    for i, condition in ipairs(conditions) do
+        scan_index = compare_conditions.get_index_for_condition(space_indexes, space_format, condition)
+
+        if scan_index ~= nil then
+            scan_iter = compare_conditions.get_tarantool_iter(condition)
+            scan_value = condition.values
+            scan_condition_num = i
+            break
+        end
+    end
+
+    -- default iteration index is primary index
+    local primary_index = space_indexes[0]
```

Comment:
this can be placed under condition?

Reply:
You mean move `base_plan.new` to `conditions.lua` file?

Reply:
File `plan.lua` was moved to `compare` folder.